### DISCUSSION TITLE
Snapdrop vs Pairdrop?

### DISCUSSION THREAD

**Discussion by ciriousjoker** [schlagmichdoch/PairDrop#115] [General]
Why did you decide to create a separate non-forked project (ie copy it) instead of improving the original project?

**Comment by schlagmichdoch** [comment]
Hi @ciriousjoker,

Thanks for the question! I understand why you are wondering about that. Let me elaborate a bit to make the decission transparent:

I tried to improve the original project:
- https://github.com/RobinLinus/snapdrop/issues/created_by/schlagmichdoch 
- https://github.com/RobinLinus/snapdrop/pulls/schlagmichdoch

None of my 10+ PR were merged or even responded to by the project owner. Not every contribution was perfect but it would have needed any  communication to make me believe that issues will be addressed and features will at least be thought about and discussed upon.

There is also this longer thread regarding maintenance: https://github.com/RobinLinus/snapdrop/issues/333
Eventhough the community helped stabilizing Snapdrop and found out part of what was wrong at the time, the project owner was not integrating us into the process at all. Mainly, we could only watch how this project more or less died as the server was constantly crashing:
https://github.com/RobinLinus/snapdrop/issues?q=is%3Aopen+is%3Aissue+down

Apparently, RobinLinus was not willing to accept any new features at all which I could somehow live with. He actually requested to focus less about new features and more about stability (see [this comment](https://github.com/RobinLinus/snapdrop/issues/333#issuecomment-1271558315)). Sadly, even after some very useful stability suggestions (e.g. [automatic restart on error](https://github.com/RobinLinus/snapdrop/issues/515#issuecomment-1365403786)) and PR regarding stability and security there was still no reaction whatsoever and there are the same problems today that were solved over half a year ago.

This was and is really frustrating.

I decided to start PairDrop by simply cherry-picking every good idea that the community already built and created PRs for and added the pairing functionality on top of that. I like that the community is actively developing this project and as community building is not possible over on the Snapdrop repo I try to do it here instead. As this project diverges from the original Snapdrop quite a bit (with [public rooms](#55 ) in the pipeline), I think it's easier for users to differentiate between the two projects by having introduced a new name.

[As I told the original creator](https://github.com/RobinLinus/snapdrop/issues/489) if he will give me a sign that he will think about merging any of my PR into Snapdrop I'm willing to create them to stabilize Snapdrop. Until then, I will not contribute to Snapdrop anymore.

Until then, I will just list every change under https://github.com/schlagmichdoch/PairDrop#differences-to-snapdrop and try to honor the original project in every way.

Regarding your question why this is not a fork: I guess this was by mistake. I already had a fork to create PRs to Snapdrop and wanted to add a second fork for PairDrop to completely separate the new development from the PR branches. Somehow, this is not natively supported by GitHub so I pushed everything using to this repo via `git push --set-upstream`. This way the complete commit history is intact but it lead to this project not being a proper fork apparently.
I don't know of a way to link the projects retrospectively. If I will ever face such a problem again, I will try to rebase everything properly so there is no need for a second branch.

**Reply by ciriousjoker** [reply]
Got it. Project owners abandoning the project is unfortunate. Would be nice to just have someone in the community maintain it. I'm sure he would have easily found someone...

I guess the original project is dead then :(
